src/mobu/main.py

```diff
@@ -47,12 +47,14 @@
 app.include_router(internal_router)
 app.include_router(external_router, prefix=f"/{config.name}")
 
+# Add middleware.
+app.add_middleware(XForwardedMiddleware)
```

Comment:
Curious to learn more about this. Is this a performance concern specific to mobu or do we need to think about this in other apps too?

Is the concern that app set up is taking longer, or that the test requests are actually running literally through multiple copies of the middleware? Can we do something about this in our middleware code?

Reply:
It's the latter; `add_middleware` doesn't check if the middleware is already added and will keep adding another copy to the middleware stack. This appears to be by design. There isn't anything we can do in our middleware code about it that I can see; we can do an early exit if we've already run, but the cost of the overhead is from all the layered calls, not from the (small) amount of work the middleware does.

I think conceptually this was my mistake in ever initializing middleware from a startup hook. It's best thought of like a router that should be included during app creation.